brain.py

```diff
@@ -1,16 +1,24 @@
-from random import randint
+import numpy as np
+from random import randint, shuffle
 from constants import *
 from game_state import State
 from mid import Mid
 from card import Card
 from hand import Hand
 
 class Brain:
-    def __init__(self, _type):
-        self.type = _type
+    def __init__(self, type=0, probability=0.5):
+        self.type = type
+        self.p = probability
+
+    def set_type(self, type):
+        self.type = type
     
-    # User's brain
-    def response0(self, state):
+    def set_probability(self, p):
+        self.p = p
```

Comment:
check if  `0 <= p <= 1` otherwise throw an exception or set some default value and print a warning